reference/forms/types/options/choice_label.rst.inc

```diff
@@ -4,11 +4,29 @@ choice_label
 .. versionadded:: 2.7
     The ``choice_label`` option was introduced in Symfony 2.7.
 
-**type**: ``string``, ``callable`` or ``false`` **default**: ``null``
+**type**: ``string``, ``callable``, :class:``Symfony\\Component\\PropertyAccess\\PropertyPath`` or ``false`` **default**: ``null``
 
 Normally, the array key of each item in the ``choices`` option is used as the
 text that's shown to the user. The ``choice_label`` option allows you to take
-more control::
+more control.
+
+If your choice values are objects (default in ``EntityType``), then ``choice_label``
+can be a string :ref:`property path <reference-form-option-property-path>`. Imagine you have some
+``Status`` class with a ``getDisplayName()`` method::
+
+    $builder->add('attending', 'choice', array(
+        'choices' => array(
+            new Status(Status::YES),
+            new Status(Status::NO),
+            new Status(Status::MAYBE),
+        ),
+        'choices_as_values' => true,
+        'choice_label' => 'displayName',
+    ));
+
+If set as a callable is called for each choice, passing you its model data ``$choice``
+and the ``$key`` from the choices array (default label). ``$value`` is either
+`choice_value`_ option value when defined or the default string value of the choice::
```

Comment:
If set as a callable, your function is called for...

... from the choices array (the default label).

And I'm not sure we need that last sentence explaining `$value`.